server/src/application/application.service.ts

```diff
@@ -11,6 +12,21 @@ export class ApplicationService {
     private applicationRepository: Repository<Application>
   ) {}
 
+  async getApplications(query: GetApplicationsQuery): Promise<Application[]> | null {
+    const where = { facilityName: Like(query.facilityName) };
```

Comment:
Could you refer to createQueryBuilder in typeorm and try using that?

Reply:
Sure thing, I pretty much made it work with the query builder, but I need to find an optimal way to sneak in that dynamic WHERE, because if it's present - it's not gonna select anything if the query string is empty.

Reply:
In createQueryBuilder, you can always have an if check and assign the where clauses! Reference https://github.com/EYDS-CA/magic-pin-trading/blob/develop/server/src/marketplace/marketplace.service.ts#L86

Reply:
Thanks Karthikk! Funny, I thought about it when I woke up. Like, what's stopping me from using just IF when it's built like this function by function. Will go ahead and fix it up, thanks for the reference!